mkt/extensions/views.py

```diff
@@ -185,3 +188,32 @@ def reject(self, request, *args, **kwargs):
         obj = self.get_object()
         obj.reject()
         return Response(status=status.HTTP_202_ACCEPTED)
+
+
+@allow_cross_site_request
+def download(request, uuid, **kwargs):
+    extension = get_object_or_404(Extension, uuid=uuid)
+
+    if extension.is_public():
+        log.info('Downloading add-on: %s from %s' % (
+                 extension.pk, extension.signed_file_path))
+        extension_etag = hashlib.sha256()
+        extension_etag.update(unicode(extension.uuid))
+        extension_etag.update(unicode(extension.file_version))
+        return get_file_response(request, extension.signed_file_path,
+                                 content_type='application/zip',
+                                 etag=extension_etag.hexdigest(), public=True)
```

Comment:
I pretty sure `content_type` or `etag` do nothing under S3 -- they have to be set on upload. We may have to do a pass looking for things like that when the files are signed.